tests/listener_test.cc

```diff
@@ -127,14 +141,42 @@ SocketWrapper bind_free_port()
 
     freeaddrinfo(servinfo); // all done with this structure
 
-    if (p == nullptr)
+    if (ai_family == AF_UNSPEC)
     {
-        fprintf(stderr, "server: failed to bind\n");
-        exit(1);
+        if (p == nullptr)
+        {
+            fprintf(stderr, "server: failed to bind\n");
+            exit(1);
+        }
+        throw std::runtime_error("failed to bind");
     }
+    
     return SocketWrapper(sockfd);
 }
 
+/*
+ * Will try to get a free port by binding port 0.
+ */
+SocketWrapper bind_free_port()
+{
+    // As of July/2024, in Linux and macOS, using AF_UNSPEC leads us to use
+    // IPv4 when available. However, in FreeBSD, it causes us to use IPv6 when
+    // available. Since Pistache itself defaults to IPv4, we try IPv4 first for
+    // bind_free_port_helper, and only try AF_UNSPEC if IPv4 fails.
```

Comment:
I'd prefer to follow the default operating system behaviour instead of forcing either IPv4 or IPv6.

Reply:
Hmmm. Yeah, I would agree with that if starting from a blank piece of paper.

I don't know where or how Pistache itself defaults to IPv4, but (per this test) it does seem to do so. Of course, I have changed the behaviour of the test to match Pistache, not the other way around.

Personally, I am reluctant to change Pistache's behaviour, given the risk that it could break code reliant on the existing behaviour.